core/client/fs/src/main/java/alluxio/client/block/stream/LocalFilePacketWriter.java

```diff
@@ -84,7 +105,20 @@ public void writePacket(final ByteBuf buf) {
 
   @Override
   public void cancel() {
-    close();
+    if (mClosed) {
+      return;
+    }
+    mClosed = true;
+
+    mCloser.register(new Closeable() {
+      @Override
+      public void close() throws IOException {
+        NettyRPC.call(mNettyRPCContext, new ProtoMessage(
+            Protocol.LocalBlockCompleteRequest.newBuilder().setBlockId(mBlockId).setCancel(true)
+                .build()));
+      }
+    });
```

Comment:
I think they are the same. It is more manageable by registering to mCloser when there are more than one places that can throw exception. Done since you prefer.